### ISSUE TITLE
Sony TV HDMI input detection

### ISSUE THREAD

**Issue by axxelh** [JeffLIrion/python-androidtv#227]
Issue [202](https://github.com/JeffLIrion/python-androidtv/issues/202) added HDMI input info, but that doesn't appear to work on current Sony Android TV implementations. 

This command appears to work on my X950H, returning just the port number (ports are 1 indexed, when the routing is 0x0000 internal apps are in use).:

'dumpsys hdmi_control | grep  -o 'mActiveRoutingPath: 0x[1-9]' | grep -o [1-9]'

There are several other fields in dumpsys that could be used, but this seemed cleanest to me.

Not sure what the right pattern is for multiple variants of CMD_HDMI_INPUT or detection of subtypes of Android TV so I'm not proposing a patch.

**Comment by JeffLIrion** [comment]
There’s a method for getting device properties:

https://github.com/JeffLIrion/python-androidtv/blob/88dab23270d2e1d8a3da3d086c08b9e06f4b4ebf/androidtv/basetv/basetv_sync.py#L185-L193

Rather than simply returning the dictionary of device properties, I think it should also store that dictionary. Then other methods could check the device properties and adjust their ADB commands accordingly, similar to how Google TV support is handled. For example, from `get_properties()`:

https://github.com/JeffLIrion/python-androidtv/blob/88dab23270d2e1d8a3da3d086c08b9e06f4b4ebf/androidtv/androidtv/androidtv_sync.py#L129

**Comment by JeffLIrion** [comment]
With the latest changes that I made, you can customize commands based on the device info. For example, see here:

https://github.com/JeffLIrion/python-androidtv/blob/e12aff4630585b4c9b1e403b4a840f4d2b739eb9/androidtv/androidtv/base_androidtv.py#L48

You could add a `_cmd_hdmi_input` attribute and customize it for applicable Sony devices.

**Comment by martomo** [comment]
@axxelh I checked out your suggested command to obtain the HDMI port number, which worked like a charm!

Although the expected `hdmi_input` value should be a hardware identifier as defined in `tv_input` module.

On my Sony Bravia KD-55XD7005 were the following hardware definitions, mapped to input using the respective TV Input Manager service intent.

* **HW2** (HDMI 1)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW2 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

* **HW3** (HDMI 2)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW3 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

* **HW4** (HDMI 3 / ARC)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW4 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

* **HW5** (HDMI 4)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW5 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

* **HW6** (AV1)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW6 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

* **HW7** (AV2)
```
am start -a android.intent.action.VIEW -d content://android.media.tv/passthrough/com.sony.dtv.tvinput.external%2F.ExternalTvInputService%2FHW7 -n com.sony.dtv.tvx/.MainActivity -f 0x10000000
```

In case you are interested, here is my proposed command, as in the linked pull request (66840c27501179288c3038ac501fdcd012cea595), to obtain hardware identifier for both Sony TV (**com.sony.dtv.tvinput.external**) and other devices (**com.mediatek.tvinput**).
```bash
dumpsys activity starter | grep -E -o '(ExternalTv|HDMI)InputService/HW[0-9]' -m 1 | grep -o 'HW[0-9]'
```

**Comment by martomo** [comment]
@axxelh @JeffLIrion This has been addressed in v0.0.58 with the changes from #236.
